### ISSUE TITLE
autosummary crashes with autosummary_generate=False

### ISSUE THREAD

**Issue by leamingrad** [sphinx-doc/sphinx#6695]
**Describe the bug**
Sphinx and autosummary are crashed if autosummary_generate is set to False.

**To Reproduce**
```
.. autosummary::
   :toctree: generated

   example.Base
```

```
autosummary_generate = False
extensions.append('sphinx.ext.autosummary')
```
```
make html
...
[autosummary] writing to _autogen
Running Sphinx v2.2.0
loading pickled environment... done

Exception occurred:
  File "/XXX/python3.7/site-packages/sphinx/ext/autosummary/__init__.py", line 726, in process_generate_options
    for genfile in genfiles]
TypeError: 'bool' object is not iterable
```

**Expected behavior**
Build successfully - possibly without autogenerated summaries.

**Your project**
None

**Screenshots**
None

**Environment info**
- OS: Debian Testing
- Python version: 3.7.3
- Sphinx version: 2.2.0
- Sphinx extensions:  sphinx.ext.autosummary

**Additional context**
This looks like a regression added by https://github.com/sphinx-doc/sphinx/pull/6506 - in particular, the test for `False` is after an `else` branch on a test for `True`.

I'm happy to push a fix for this, but I just want to my diagnosis is correct before doing so.

**Comment by leamingrad** [comment]
Note that setting `autosummary = []` is a workaround for this

**Comment by tk0miya** [comment]
I already fixed this in #7365. So closing now. Thank you for reporting.
